### ISSUE TITLE
Overflow menu is not (reasonably) navigable to screenreaders

### ISSUE THREAD

**Issue by hvlawren** [carbon-design-system/carbon#823]
>Describe in detail the issue you're having. Is this a feature request (new component, new icon), a bug, or a general issue?

When the overflow menu is not the last element of the DOM, the overflow menu is not reasonably accessible. On the overflow menu codepen, it appears to be, as there are not other elements. On the DataTable codepen, this is apparent

> Is this issue related to a specific component? 

Overflow menu, more apparent in the DataTable overflow menu

> What did you expect to happen? What happened instead? What would you like to see changed?

When a screen reader user selects the overflow menu, the options should be navigable next

> What browser are you working in? 

Safari with VoiceOver

> What version of the Carbon Design System are you using? 

carbondesignsystem.com current version

## Steps to reproduce the issue

1. Go to https://codepen.io/team/carbon/full/MOEagV/
2. Navigate to the overflow menu of the first row
3. Select the overflow menu
4. Advance to the next element

The first element of the next row is read rather than the overflow menu itself.

This is because the menu is not in correct DOM order. Ideally, the menu would come directly after the toggle on the DOM.

I'm less sure on this next part, but I believe the button should also have aria-controls, aria-expanded, and aria-haspopup attributes. I could be mistaken on the attributes.

I'm by no means an expert at using a screenreader, so I may be mistaken, however, this seems like an issue.

**Comment by hvlawren** [comment]
Note that if instead of the first overflow menu, you toggle the last overflow menu, the UX seems reasonable, as you navigate to the menu after selecting the toggle button.

**Comment by asudoh** [comment]
Ref: https://github.com/carbon-design-system/carbon-components-react/pull/518

**Comment by carmacleod** [comment]
See also #666

**Comment by asudoh** [comment]
I think the author of this issue meant https://codepen.io/team/carbon/pen/MOEwjp/ CodePen, and I see the problem has been fixed with `<input class="bx--visually-hidden" type="text">` focus trap. Let us know if anybody still see a problem. Thanks!

**Comment by hvlawren** [comment]
No I specified the correct codepen -- unless the version of Carbon is out of date on the one I specified.

When navigating to the overflow menu button of the first view and selecting it, when you advance to the next focusable item, the actual menu is not the next item, it advances to the next row of the table.

**Comment by hvlawren** [comment]
I would recommend this issue be reopened unless the codepen I specified is out of date.

**Comment by asudoh** [comment]
@hvlawren Updated 
https://codepen.io/team/carbon/full/MOEagV/ - My apologies for naïvely assuming it’s just about data table.
